Results- Generative Model Random Sampling (#13)
greenelab/text_mined_hetnet_manuscript: content/04.results.md

## Results

### Generative Model Using Randomly Sampled Label Functions

> To evaluate the feasibility of label function reuse we used a random sampling method.

Reviewer: ```suggestion
We added randomly sampled label functions to a baseline for each edge type to evaluate the feasibility of label function re-use.
```

> Our baseline model consisted of training the generative model with only distant supervision label functions for each edge type.

Reviewer: ```suggestion
Our baseline model consisted of a generative model trained with only the edge type's distant supervision label functions.
```

> We report the results in the form of area under the precision recall curve (AUPRC) (Figure {@fig:gen_model_auprc}) and area under the receiver operating curve (AUROC) (Figure {@fig:gen_model_auroc}).  

Reviewer 2: See https://github.com/greenelab/deep-review/issues/613... if you want consistency you can do:

+ AUROC and AUPR
+ AUROCC and AUPR

> The general trend is performance increases as the number of edge specific label functions are added on top of the edge specific baseline models.

Reviewer: ```suggestion
The on-diagonal plots of figure {@fig:gen_model_auprc}) and figure {@fig:gen_model_auprc} show performance when edge-specific label functions are added on top of edge-specific baselines.
The general trend is performance increases in this setting.
```

> This trend is shown on the on-diagonals of figure {@fig:gen_model_auprc}) and figure {@fig:gen_model_auprc}.

Reviewer: ```suggestion
```

> A quintessential example of this trend is with the compound treats disease (CtD) edge type.

Reviewer: ```suggestion
The compound treats disease (CtD) edge type is a quintessential example of this trend.
```

> The baseline model starts off with an auroc score of 0.52 and an auprc of 0.28, then increases to 0.76 and 0.49 as the number of edge specific label functions are incorporated. 

Reviewer 2: Could do 28% rather than 0.28. Maybe a bit easier to read

> The baseline model starts off with an auroc score of 52% and an auprc of 28%, then increases to 76% and 49% as the number of edge specific label functions are incorporated. 

Reviewer: ```suggestion
The baseline model starts off with an AUROC score of 52% and an AUPRC of 28%, which increase to 76% and 49% respectively as more CtD label functions are included. 
```

> The disease associates gene (DaG) edge has a similar trend with performance starting off with a auroc score of 56% and auprc score of 41% then increasing to 62% and 45% respectively.

Reviewer: ```suggestion
Disease associates gene (DaG) edges have a similar trend: performance starting off with a AUROC of 56% and AUPRC of 41%, which increase to 62% and 45% respectively.
```

> Both the Compound binds Gene (CbG) and Gene interacts Gene (GiG) edges have an increasing trend (auroc of 78% to 82% and 66% to 77% respectively), but plateaus as the number of label functions increases.  

Reviewer: ```suggestion
Both the Compound binds Gene (CbG) and Gene interacts Gene (GiG) edges have an increasing trend but plateau after a few label functions are added.  
```

> For the off-diagonals performance varies upon edge type as some edges can aide in predicting others.

Reviewer: ```suggestion

The off-diagonals in figure {@fig:gen_model_auprc}) and figure {@fig:gen_model_auprc} show how performance varies when label functions from one edge type are added to a different edge type's baseline.
```

> Looking at the DaG row, performance increases regardless of the edge type source.

Reviewer: ```suggestion
In certain cases (apparent for DaG), performance increases regardless of the edge type used for label functions.
```

> This trend changes for CtD as performance starts to increase, when only adding one label function, then plateaus (auroc) or decreases (auprc) when more label functions are added.

Reviewer: ```suggestion
In other cases (apparent with CtD), one label function appears to improve performance; however, adding more label functions does not improve performance (AUROC) or decreases it (AUPRC).
```

> For CbG performance decreases when using label functions from the DaG and CtD category.

Reviewer: ```suggestion
In certain cases, the source of the label functions appear to be important: for CbG edges performance decreases when using label functions from the DaG and CtD categories.
```

> As expected performance increases for both auprc and auroc when using label functions from the GiG edge type.

Reviewer: ```suggestion

Our initial hypothesis was based on the idea that certain edge types capture similar physical relationships and that these cases would be particularly amenable for label function transfer.
For example, compound-binds-gene (CbG) and gene-interacts-gene (GiG) both describe physical interactions.
We observed that performance increased as assessed by both AUPRC and AUPRC when using label functions from the GiG edge type to predict CbG edges.
```

> This trend is shared when predicting the GiG edge; however the magnitude of performance is low.  

Reviewer: ```suggestion
A similar trend was observed when predicting the GiG edge; however, the performance differences were small for this edge type making the importance difficult to assess.  
```

> Lastly, performance increases (auroc and auprc) for both DaG and CtD, when sampling from all label functions at once.

Reviewer: ```suggestion
The last column shows performance when sampling from all label functions.
Performance increased (AUROC and AUPRC) for both DaG and CtD, when sampling from the full pool of label functions.
```

> Interestingly, CbG and GiG share a similar trend of increasing in performance when one random label function is used, but drastically decreases as more label functions are incorporated.

Reviewer: ```suggestion
CbG and GiG also had increased performance when one random label function was sampled, but performance decreased drastically as more label functions were added.
It is possible that a small number of irrelevant label functions are able to overwhelm the distant supervision label functions in these cases.
```

Reviewer: Adding the last sentence here, though it's a bit more of a discussion section point so we may move it once the first set of PRs are merged.

> ![

Reviewer: You may want to move this figure to earlier in the section so that there's some up front figures for folks to look at.
